npm_and_yarn/lib/dependabot/npm_and_yarn/file_updater/npmrc_builder.rb

```diff
@@ -150,6 +153,16 @@ def credential_lines_for_npmrc
           ["always-auth = true"] + lines
         end
 
+        def npmrc_scoped_registries
+          return [] unless npmrc_file
+
+          @npmrc_scoped_registries ||=
+            npmrc_file.content.lines.select { |line| line.match?(/^\s*@/) && line.include?(":registry") }.
+            map do |line|
+              line.match(/:registry\s*=\s*(?<registry>\S+)/)&.named_captures&.fetch("registry")
```

Comment:
We could maybe extract a constant for the regex here, but not blocking for review